tests/core/data_model/test_dataobject.py

```diff
@@ -878,6 +878,51 @@ class Derived(Base):
     assert inst.baz == "qwer"
 
 
+def test_dataobject_union_repeated():
+    """Make sure that a oneof with lists of primitive fields works correctly"""
+
+    # convert
+    @dataobject
+    class Foo(DataObjectBase):
+        foo: Union[List[str], List[int]]
+
+    # The above behaves the same way as this:
+
+    # @dataobject
+    # class Foo(DataObjectBase):
+    #     @dataobject
+    #     class IntSequence(DataObjectBase):
+    #         values: List[int]
+
+    #     @dataobject
+    #     class StrSequence(DataObjectBase):
+    #         values: List[str]
+
+    #     foo: Union[IntSequence, StrSequence]
+
+    # proto round trip
+    foo_int = Foo.IntSequence(values=[1, 2])
+    foo1 = Foo(foo=foo_int)
```

Comment:
Is this the only way these can be constructed?

It's good, but I think a native python @dataclass user would expect to be able to write
```
foo_with_ints = Foo([1, 2, 3])
foo_with_strs = Foo(["1", "2", "3"])
```

Reply:
Hmm, yeah that would be ideal, but can that be a separate feature request? This PR is mostly to handle the lists within function arguments, I believe in order to support the above, there _might_ have to be metaclass changes? Not sure